### PR TITLE
Large Pages on Linux & macOS

### PR DESCRIPTION
**Pull request by mjsabby**
Part of #18371.

This change adds Large Pages support for Linux (embedded GC, standalone was done in a previous change) and macOS (both embedded GC and standalone).

### CODE REVIEW

**Comment by mjsabby** [comment]
cc @Maoni0 @janvorli

File: src/pal/src/map/virtual.cpp [hunk 1]
```diff
@@ -2145,7 +2160,7 @@ void ExecutableMemoryAllocator::TryReserveInitialMemory()
     // Do actual memory reservation.
     do
     {
-        m_startAddress = ReserveVirtualMemory(pthrCurrent, (void*)preferredStartAddress, sizeOfAllocation);
+        m_startAddress = ReserveVirtualMemory(pthrCurrent, (void*)preferredStartAddress, sizeOfAllocation, 0);
```
Comment (janvorli) [inline src/pal/src/map/virtual.cpp]: A nit - can you please add comment on the last argument (like `0 /* fAllocationType */ `) here and at the other place below?

**Comment by mjsabby** [comment]
I think the nt x86 checked build is flaky. Is there a way to only retrigger that?

**Comment by janvorli** [comment]
> Is there a way to only retrigger that?

Yes, click on the details and then on the left, you'll see links to re-run the specific test, re-run all failed tests, etc.

**Comment by mjsabby** [comment]
I don’t see such a link when I click details. I guess I’ll have to repush to trigger.

**Comment by mjsabby** [comment]
```Return code:      1
Raw output file:      C:\\dotnetbuild\\work\\16e30220-03a3-4fe3-8e6d-58340d696cf2\\Work\\48818002-c712-4708-a55b-4eb594d5db99\\Exec\\Interop\\COM\\Reports\\Interop.COM\\NativeClients\\DefaultInterfaces\\DefaultInterfaces.output.txt
Raw output:
BEGIN EXECUTION
Copying 'C:\\dotnetbuild\\work\\16e30220-03a3-4fe3-8e6d-58340d696cf2\\Payload\\CoreShim.dll'...
        1 file(s) copied.
 \"C:\\dotnetbuild\\work\\16e30220-03a3-4fe3-8e6d-58340d696cf2\\Payload\\corerun.exe\" DefaultInterfaces.exe 
Searching for exe to launch in C:\\dotnetbuild\\work\\16e30220-03a3-4fe3-8e6d-58340d696cf2\\Work\\48818002-c712-4708-a55b-4eb594d5db99\\Exec\\Interop\\COM\\NativeClients\\DefaultInterfaces...
Launching 'C:\\dotnetbuild\\work\\16e30220-03a3-4fe3-8e6d-58340d696cf2\\Work\\48818002-c712-4708-a55b-4eb594d5db99\\Exec\\Interop\\COM\\NativeClients\\DefaultInterfaces\\COMClientDefaultInterfaces.exe'...
Activate class using default interfaces via IUnknown...
Fail to activate class via a default interface...
FALSE: hr == COR_E_INVALIDOPERATION
Test Failure: 0x80004005```

@MichalStrehovsky is this test flaky?

**Comment by MichalStrehovsky** [comment]
> @MichalStrehovsky is this test flaky?

That test is new as of #23970. I have a suspicion the test binaries don't match product binaries that are tested. I've seen something similar this week...

**Comment by mjsabby** [comment]
@MichalStrehovsky but that's the whole point of testing ... Ok. I guess I'll rerun all the tests again.

**Pull request merged**